src/database/schema/ChatSchema.ts

```diff
@@ -267,15 +268,26 @@ class ChatSchema implements BaseDbSchema {
     channelId: string,
     message: string,
     localDb: SQLiteDatabase,
-    type: 'regular' | 'system' = 'regular',
-    status: 'pending' | 'sent' = 'pending'
+    type: 'regular' | 'system' | 'reply' = 'regular',
+    status: 'pending' | 'sent' = 'pending',
+    json: string | null = null
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `json` parameter and its subsequent usage in the function is a good improvement as it allows for more flexibility when creating a new chat message. However, it's important to handle the case where `json` might be `null` or an invalid JSON string. The current implementation does handle this by setting `rawJson` to `null` if `JSON.stringify(json)` throws an error, which is a good practice.
```diff
-    status: 'pending' | 'sent' = 'pending'
+    status: 'pending' | 'sent' = 'pending',
+    json: string | null = null
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->